Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -168,7 +169,16 @@ class PaintedContentsInfo {
 #if HAVE(HDR_SUPPORT)
     bool paintsHDRContent()
     {
-        determinePaintsContent();
+        if (m_hdrContent == RequestState::Unknown)
+            determinePaintsContent();
+        return m_hdrContent == RequestState::True || m_hdrContent == RequestState::Undetermined;
```

Comment:
Is it right to say that we need HDR if the state is "Undetermined"? That means that every layer with a huge layer tree will use HDR.

Reply:
Checking `RequestState::Undetermined` was removed. In fact `m_hdrContent` is never set to `Undetermined`.